BUG: plot_io_cost() handling of a mostly STDIO log file

I wasn't actually looking to trigger this, but just so it is bookmarked I'll put some details below. It probably is lower priority than what I was actually looking into because I think this is mostly just "nearly-empty log cornercase."

For C file:

```c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char** argv) {
    MPI_Init(NULL, NULL);
    int world_rank;
    MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);
    if (world_rank % 2 == 0) {
        printf("Hello world\n");
    }
    MPI_Finalize();
}
```

Working with `darshan` project (runtime and util/python bindings) at hash 548d9aed64, and producing log file in this tarball:
[simple_stdio_log_problem.tar.gz](https://github.com/darshan-hpc/darshan/files/8642434/simple_stdio_log_problem.tar.gz) using:
```
mpicc issue_730.c
mpirun -x DXT_ENABLE_IO_TRACE=1 -x DARSHAN_LOGPATH=/vast/home/treddy/rough_work/darshan/issue_730 -x DARSHAN_DISABLE_SHARED_REDUCTION=1 -x LD_PRELOAD=/vast/home/treddy/darshan_install/lib/libdarshan.so -np 40 ./a.out
```


The traceback below the fold is produced when running the full pydarshan test suite with the above log file inserted into the logs repo locally. Maybe we can handle this a bit more gracefully at some point.

<details>

```
========================================================================================================================================================= FAILURES =========================================================================================================================================================
__________________________________________ test_main_all_logs_repo_files[/vast/home/treddy/virtualenvs/py39_skylake_gold_darshan/lib/python3.9/site-packages/darshan_logs/runtime_heatmap_inactive_ranks/treddy_a.out_id13734580-585841_5-6-39231-12369746513475705069_1.darshan] __________________________________________
[gw31] linux -- Python 3.9.5 /vast/home/treddy/virtualenvs/py39_skylake_gold_darshan/bin/python

tmpdir = local('/ram/tmp/pytest-of-treddy/pytest-1/popen-gw31/test_main_all_logs_repo_files_0'), log_filepath = '/vast/home/treddy/virtualenvs/py39_skylake_gold_darshan/lib/python3.9/site-packages/darshan_logs/runtime_heatmap_inactive_ranks/treddy_a.out_id13734580-585841_5-6-39231-12369746513475705069_1.darshan'

    @pytest.mark.skipif(not pytest.has_log_repo,
                        reason="missing darshan_logs")
    @pytest.mark.parametrize("log_filepath",
            _provide_logs_repo_filepaths()
            )
    def test_main_all_logs_repo_files(tmpdir, log_filepath):
        # similar to `test_main_without_args` but focused
        # on the Darshan logs from the logs repo:
        # https://github.com/darshan-hpc/darshan-logs
    
        if "e3sm_io_heatmap_and_dxt" in log_filepath:
            pytest.xfail(reason="large memory requirements")
        argv = [log_filepath]
        with mock.patch("sys.argv", [""] + argv):
            with tmpdir.as_cwd():
                # generate the summary report
>               summary.main()

/vast/home/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/tests/test_summary.py:163: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/vast/home/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py:569: in main
    report_data = ReportData(log_path=log_path)
/vast/home/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py:135: in __init__
    self.register_figures()
/vast/home/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py:425: in register_figures
    io_cost_fig = ReportFigure(**io_cost_params)
/vast/home/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py:73: in __init__
    self.generate_fig()
/vast/home/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py:100: in generate_fig
    fig = self.fig_func(**self.fig_args)
/vast/home/treddy/github_projects/darshan/darshan-util/pydarshan/darshan/experimental/plots/plot_io_cost.py:154: in plot_io_cost
    io_cost_df.plot.bar(stacked=True, rot=0, ax=ax_raw, legend=False, zorder=3)
/vast/home/treddy/virtualenvs/py39_skylake_gold_darshan/lib/python3.9/site-packages/pandas/plotting/_core.py:1131: in bar
    return self(kind="bar", x=x, y=y, **kwargs)
/vast/home/treddy/virtualenvs/py39_skylake_gold_darshan/lib/python3.9/site-packages/pandas/plotting/_core.py:972: in __call__
    return plot_backend.plot(data, kind=kind, **kwargs)
/vast/home/treddy/virtualenvs/py39_skylake_gold_darshan/lib/python3.9/site-packages/pandas/plotting/_matplotlib/__init__.py:71: in plot
    plot_obj.generate()
/vast/home/treddy/virtualenvs/py39_skylake_gold_darshan/lib/python3.9/site-packages/pandas/plotting/_matplotlib/core.py:327: in generate
    self._compute_plot_data()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pandas.plotting._matplotlib.core.BarPlot object at 0x146d03181b20>

    def _compute_plot_data(self):
        data = self.data
    
        if isinstance(data, ABCSeries):
            label = self.label
            if label is None and data.name is None:
                label = "None"
            if label is None:
                # We'll end up with columns of [0] instead of [None]
                data = data.to_frame()
            else:
                data = data.to_frame(name=label)
        elif self._kind in ("hist", "box"):
            cols = self.columns if self.by is None else self.columns + self.by
            data = data.loc[:, cols]
    
        # GH15079 reconstruct data if by is defined
        if self.by is not None:
            self.subplots = True
            data = reconstruct_data_with_by(self.data, by=self.by, cols=self.columns)
    
        # GH16953, _convert is needed as fallback, for ``Series``
        # with ``dtype == object``
        data = data._convert(datetime=True, timedelta=True)
        include_type = [np.number, "datetime", "datetimetz", "timedelta"]
    
        # GH23719, allow plotting boolean
        if self.include_bool is True:
            include_type.append(np.bool_)
    
        # GH22799, exclude datetime-like type for boxplot
        exclude_type = None
        if self._kind == "box":
            # TODO: change after solving issue 27881
            include_type = [np.number]
            exclude_type = ["timedelta"]
    
        # GH 18755, include object and category type for scatter plot
        if self._kind == "scatter":
            include_type.extend(["object", "category"])
    
        numeric_data = data.select_dtypes(include=include_type, exclude=exclude_type)
    
        try:
            is_empty = numeric_data.columns.empty
        except AttributeError:
            is_empty = not len(numeric_data)
    
        # no non-numeric frames or series allowed
        if is_empty:
>           raise TypeError("no numeric data to plot")
E           TypeError: no numeric data to plot

/vast/home/treddy/virtualenvs/py39_skylake_gold_darshan/lib/python3.9/site-packages/pandas/plotting/_matplotlib/core.py:506: TypeError
```

</details>

--- a/darshan-util/pydarshan/darshan/cli/summary.py
+++ b/darshan-util/pydarshan/darshan/cli/summary.py
@@ -332,6 +332,22 @@ class ReportData:
         """
         self.figures = []
 
+        if not self.report.modules:
+            # no data in report to summarize, print warning and that's it
+            no_data_message = (
+                "This Darshan log file has no instrumentation records,"
+                "there is no data to plot. Did this app do any I/O?"
+            )
+            fig = ReportFigure(
+                section_title="",
+                fig_title="",
+                fig_func=None,
+                fig_args=None,
+                fig_description=no_data_message,
+            )
+            self.figures.append(fig)
+            return
+
         #########################################
         ## Add the runtime and/or DXT heat map(s)
         #########################################

--- a/darshan-util/pydarshan/darshan/tests/test_summary.py
+++ b/darshan-util/pydarshan/darshan/tests/test_summary.py
@@ -209,9 +209,14 @@ def test_main_all_logs_repo_files(tmpdir, log_filepath):
                             stdio_position > -1):
                             assert mpiio_position < posix_position < stdio_position
                     else:
-                        # check that help message is present
-                        assert "Heatmap data is not available for this job" in report_str
-                        assert "Consider enabling the runtime heatmap module" in report_str
+                        if not "empty_log" in log_filepath:
+                            # check that help message is present
+                            assert "Heatmap data is not available for this job" in report_str
+                            assert "Consider enabling the runtime heatmap module" in report_str
+                        else:
+                            # check empty log warning and return
+                            assert "This Darshan log file has no instrumentation records" in report_str
+                            return
 
                     # check if I/O cost figure is present
                     for mod in report.modules:

